src/components/Shared/ProposalDescription.tsx

```diff
@@ -36,21 +36,25 @@ export default class ProposalDescription extends React.Component<IExternalProps>
 
   private renderDescription = (props: { href: string; children: React.ReactNode }) => {
     if (props.href) {
-      const url = new URL(props.href);
-      const videoId = this.parseYouTubeVideoIdFromUri(props.href);
-      if (videoId) {
-        const start = url.searchParams.get("t") || "0";
-
-        return <iframe className={css.embeddedVideo} frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen
-          src={`${url.protocol}//www.youtube-nocookie.com/embed/${videoId}?start=${start}`}>
-        </iframe>;
-      } else {
-        const videoId = this.getVimeoIdFromUrl(props.href);
+      try {
+        const url = new URL(props.href);
+        const videoId = this.parseYouTubeVideoIdFromUri(props.href);
         if (videoId) {
-          return <iframe className={css.embeddedVideo} frameBorder="0" allow="autoplay; fullscreen" allowFullScreen
-            src={`${url.protocol}//player.vimeo.com/video/${videoId}`}>
+          const start = url.searchParams.get("t") || "0";
+
+          return <iframe className={css.embeddedVideo} frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen
+            src={`${url.protocol}//www.youtube-nocookie.com/embed/${videoId}?start=${start}`}>
           </iframe>;
+        } else {
+          const videoId = this.getVimeoIdFromUrl(props.href);
+          if (videoId) {
+            return <iframe className={css.embeddedVideo} frameBorder="0" allow="autoplay; fullscreen" allowFullScreen
+              src={`${url.protocol}//player.vimeo.com/video/${videoId}`}>
+            </iframe>;
+          }
         }
+      } catch (err) {
+        console.error(err);
```

Comment:
Can you please add a comment to the code:

```
/**
  *  This is particularly to trap an exception thrown by `new URL` when an href is not a valid URL, for example, an email address"
 **/
```

Reply:
added